File: AgeSavings.py
import datetime
from datetime import date
from dateutil.relativedelta import relativedelta

def convert(toDate):
    format = '%d-%m-%Y'
    dateFormat = datetime.datetime.strptime(toDate, format)
    dateFormat = dateFormat.date()
    return dateFormat

def numOfDays(date1, date2) :
    return (date1-date2).days

def numOfYears(date1, date2) :
    return relativedelta(date1, date2).years

dob = input("\n Enter your Date of Birth (in dd-MM-yyyy format) : ")
dobDate = convert(dob)
birthYear = dobDate.year

currDate = date.today()
currYear = currDate.year
numofDaysOld = numOfDays(currDate, dobDate)
numofYearsOld = numOfYears(currDate, dobDate)
loopcounter = 1
savings = 0
yearCounter = birthYear

if((birthYear % 400 == 0) or ((birthYear % 4 == 0) and (birthYear % 100 != 0))) :
            savings = 366
else :
            savings = 365

#for loopcounter in range(numofYearsOld,0,-1) :

#    if((yearCounter % 400 == 0) or ((yearCounter % 4 == 0) and (yearCounter % 100 != 0))) :
#        savings = savings + (loopcounter*366)
#    else :
#        savings = savings + (loopcounter*365)

while(loopcounter<numofYearsOld) :
    if((yearCounter % 400 == 0) or ((yearCounter % 4 == 0) and (yearCounter % 100 != 0))) :
        savings = savings + (loopcounter*366)
    else :
        savings = savings + (loopcounter*365)
    loopcounter += 1
    
print("\n You are ",numofDaysOld," Days Old!")
print("\n You are ",numofYearsOld," Years Old!")
print("\n Your Total estimated Savings would be : Rs.",savings)

File: RealTimeExamples/AgeSavings.java
package RealTimeExamples;

import java.io.*;
import java.util.*;
import java.util.Date;
import java.util.concurrent.TimeUnit;
import java.text.ParseException;
import java.text.SimpleDateFormat;
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.time.format.DateTimeFormatter;
import java.time.temporal.ChronoUnit;

public class AgeSavings {

    static int daysDifference(String date1, String date2) {
        DateTimeFormatter dtf = DateTimeFormatter.ofPattern("dd-MM-yyyy");
        long daysDiff = 0L;

        try {
            LocalDate d1 = LocalDate.parse(date1, dtf);
            LocalDate d2 = LocalDate.parse(date2, dtf);

            daysDiff = ChronoUnit.DAYS.between(d1, d2);
        }
        catch (Exception e) {
            e.printStackTrace();
        }

        int result = (int)daysDiff;
        return result;
    }

    static int yearsDifference(String date1, String date2) {
        DateTimeFormatter dtf = DateTimeFormatter.ofPattern("dd-MM-yyyy");
        long yearsDiff = 0L;

        try {
            LocalDate d1 = LocalDate.parse(date1, dtf);
            LocalDate d2 = LocalDate.parse(date2, dtf);

            yearsDiff = ChronoUnit.YEARS.between(d1, d2);
        }
        catch (Exception e) {
            e.printStackTrace();
        }

        int result = (int)yearsDiff;
        return result;
    }

    
    public static void main(String[] args) {

        DateTimeFormatter dtf = DateTimeFormatter.ofPattern("dd-MM-yyyy");
        System.out.println("\n Enter your Date of Birth (in dd-MM-yyyy format) : ");
        Scanner dobInput = new Scanner(System.in);
        String dob = dobInput.next();
        LocalDate dobDateFormat = LocalDate.parse(dob, dtf);
        
        int year = dobDateFormat.getYear();
        System.out.println("\n Birth Year : "+year);
        
        LocalDate currDateTime = LocalDate.now();
        String todaydate = currDateTime.format(dtf);
        
        int noofDaysOld = daysDifference(dob, todaydate);
        int noofYearsOld = yearsDifference(dob, todaydate);
                
        int savings = 0;

        if((year % 400 == 0) || ((year % 4 == 0) && (year % 100 != 0))) {
            savings = 366;
        } else {
            savings = 365;
        }

        for (int counter=1; counter<noofYearsOld; counter++) {

            if((year % 400 == 0) || ((year % 4 == 0) && (year % 100 != 0))) {
                savings = savings + (counter*366);
            }
            else {
                savings = savings + (counter*365);
            }
            year++;

        }

        System.out.println("\n You are "+noofDaysOld+" Days Old!");
        System.out.println("\n You are "+noofYearsOld+" Years Old!");
        System.out.println("\n Your Total estimated Savings would be : Rs."+savings);
        
        dobInput.close();
    }
}
